app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklySent.java

```diff
@@ -0,0 +1,70 @@
+package nyc.c4q.jonathancolon.inContaq.utlities.sms.weekly;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import org.joda.time.DateTime;
+import java.util.ArrayList;
+import java.util.TreeMap;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+
+public class WeeklySent extends AsyncTask<WeeklyTask, Void, TreeMap<Integer, Integer>> {
+
+    private TreeMap<Integer, Integer> weeklyTexts;
+
+    public WeeklySent() {
+    }
+
+    @Override
+    protected void onPreExecute() {
+        Log.i("WeeklySentTask", "Getting Weekly Total Sent...");
+    }
+
+    @Override
+    protected TreeMap<Integer, Integer> doInBackground(WeeklyTask... params) {
+
+        ArrayList<Sms> listSms = params[0].listSms;
+        weeklyTexts = params[0].weeklyTexts;
+        return getSmsStats(listSms);
+
+    }
+
+    @Override
+    protected void onPostExecute(TreeMap<Integer, Integer> ret) {
+        super.onPostExecute(ret);
+    }
+
+    private TreeMap<Integer, Integer> getSmsStats(ArrayList<Sms> list) {
+
+        ArrayList<String> sentSms = new ArrayList<>();
```

Comment:
List<String> sentSms = new ArrayList<>(); 
get in the habit of declaring by interface.